base/env.jl

```diff
@@ -7,8 +7,10 @@ if Sys.iswindows()
     _hasenv(s::Vector{UInt16}) = _getenvlen(s) != 0 || Libc.GetLastError() != ERROR_ENVVAR_NOT_FOUND
     _hasenv(s::AbstractString) = _hasenv(cwstring(s))
 
+    const JULIA_DEBUG_CWSTRING = cwstring("JULIA_DEBUG")
```

Comment:
Instead of special-casing `JULIA_DEBUG` to solve the performance issue in this one known instance, is it a stupid idea to fill a dict with those precomputed cwstrings for all "known" (e.g. in Base, or commonly used) env vars, so that access will be faster for all those cases? If such a list exists or can easily be compiled, that is.

Reply:
I think an entirely sensible thing to do for them is to store every query in an IdDict, so they only get computed the first time

Reply:
Alternatively, we could use the libuv API for all platforms, which includes the string conversion internally